pkg/codegen/internal/test/testdata/simple-resource-schema/nodejs/resource.ts

```diff
@@ -51,6 +51,13 @@ export class Resource extends pulumi.CustomResource {
         if (!opts.version) {
             opts = pulumi.mergeOptions(opts, { version: utilities.getVersion()});
         }
+        // Always mark these fields as secret to avoid leaking sensitive values into the state.
+        for (const key of ["bar"]) {
+            if (key in inputs && inputs[key] !== undefined) inputs[key] = pulumi.secret(inputs[key]);
+        }
```

Comment:
This feels like a slightly unexpected place and way to do this. 

A few lines higher - we assign into `inputs["bar"]`. Should we just apply this `secret()` wrapping directly as part of that existing assignment?

Reply:
I moved this to the assignment, but it also required using optional chaining to avoid the case of `pulumi.secret(undefined)`.

It looks like this now: `inputs["data"] = args?.data ? pulumi.secret(args.data) : undefined;`